samtranslator/model/eventsources/push.py

```diff
@@ -47,9 +48,13 @@ def _construct_permission(self, function, source_arn=None, source_account=None,
         :returns: the permission resource
         :rtype: model.lambda_.LambdaPermission
         """
-        lambda_permission = LambdaPermission(self.logical_id + 'Permission' + suffix,
+        if suffix.isalnum():
+            permission_logical_id = self.logical_id + 'Permission' + suffix
+        else:
+            generator = logical_id_generator.LogicalIdGenerator(self.logical_id + 'Permission', suffix)
+            permission_logical_id = generator.gen()
```

Comment:
The difficulty with changing the Lambda permission naming here is that it will be difficult to reference the function permissions that are created. I think that this will be fine, as nothing should depend on those resources.

Could you make a note on https://github.com/awslabs/serverless-application-model/blob/master/docs/internals/generated_resources.rst#api that if the `StageName` parameter has a value that contains non alpha-numeric characters, the generated names will follow this new convention?